In [1]:
# import logging
# import os
# import pdb
# import numpy as np
# import pandas as pd
# import utils
# import time
# import warnings
# import csv
# import os
import during_crossfit_params
import before_crossfit_params_new

In [2]:
ci_estimator_list = ['tl', 'dml', 'kernel_dml', 'CausalForestDML']
model_y = ['linear', 'forest', 'gbf', 'nnet']
model_t = ['linear', 'forest', 'gbf', 'nnet']

IHDP Dataset

During Cross-fitting

In [ ]:
key = 'ihdp'
during_crossfit_params.main(ci_estimator_list, model_y, model_t, key, k=4)

{'fold 0': {'model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'model_t': LogisticRegressionCV(Cs=1, max_iter=25), 'Mse': {'tl': 7.522415610112231, 'dml': 6.1958663124037665, 'kernel_dml': 5.092179307704854, 'CausalForestDML': 6.104313206506676}, 'Runtime': {'tl': 0.15604853630065918, 'dml': 9.630453824996948, 'kernel_dml': 0.4279961585998535, 'CausalForestDML': 0.5730080604553223}}, 

'fold 1': {'model_y': RandomForestRegressor(max_depth=10, n_estimators=25), 'model_t': RandomForestClassifier(max_depth=5, min_samples_leaf=2, min_samples_split=5, n_estimators=25), 'Mse': {'tl': 8.629223154813356, 'dml': 8.14990031417063, 'kernel_dml': 5.236557810041166, 'CausalForestDML': 5.726556547598269}, 'Runtime': {'tl': 0.14014148712158203, 'dml': 1.2774248123168945, 'kernel_dml': 0.319002628326416, 'CausalForestDML': 0.4391043186187744}}, 

'fold 2': {'model_y': GradientBoostingRegressor(), 'model_t': GradientBoostingClassifier(learning_rate=0.01), 'Mse': {'tl': 8.653218183975012, 'dml': 7.229983065805759, 'kernel_dml': 4.545099065854809, 'CausalForestDML': 5.69423235994597}, 'Runtime': {'tl': 0.14896464347839355, 'dml': 2.4197700023651123, 'kernel_dml': 0.4591255187988281, 'CausalForestDML': 0.5746400356292725}}, 

'fold 3': {'model_y': MLPRegressor(max_iter=25), 'model_t': MLPClassifier(alpha=0.001, hidden_layer_sizes=(50,), max_iter=25), 'Mse': {'tl': 17.623448729826627, 'dml': 9.160365341698474, 'kernel_dml': 4.49084190576368, 'CausalForestDML': 6.753502026422641}, 'Runtime': {'tl': 0.2607533931732178, 'dml': 0.647728443145752, 'kernel_dml': 0.4145948886871338, 'CausalForestDML': 0.5427343845367432}}} 

46.24497675895691


{'tl': [7.522415610112231, 8.629223154813356, 8.653218183975012, 17.623448729826627],
'dml': [6.1958663124037665, 8.14990031417063, 7.229983065805759, 9.160365341698474],
'kernel_dml': [5.092179307704854, 5.236557810041166, 4.545099065854809, 4.49084190576368],
'CausalForestDML': [6.104313206506676, 5.726556547598269, 5.69423235994597, 6.753502026422641]}

{'tl': 0, 'dml': 0, 'kernel_dml': 3, 'CausalForestDML': 2}

{'tl': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': '', 'Mse': 7.522415610112231},
'dml': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(Cs=1, max_iter=25), 'Mse': 6.1958663124037665},
'kernel_dml': {'best_model_y': MLPRegressor(max_iter=25), 'best_model_t': MLPClassifier(alpha=0.001, hidden_layer_sizes=(50,), max_iter=25), 'Mse': 4.49084190576368},
'CausalForestDML': {'best_model_y': GradientBoostingRegressor(), 'best_model_t': GradientBoostingClassifier(learning_rate=0.01), 'Mse': 5.69423235994597}}


{'tl': [0.15604853630065918, 0.14014148712158203, 0.14896464347839355, 0.2607533931732178], 
'dml': [9.630453824996948, 1.2774248123168945, 2.4197700023651123, 0.647728443145752], 
'kernel_dml': [0.4279961585998535, 0.319002628326416, 0.4591255187988281, 0.4145948886871338], 
'CausalForestDML': [0.5730080604553223, 0.4391043186187744, 0.5746400356292725, 0.5427343845367432]} 

{'tl': 1, 'dml': 3, 'kernel_dml': 1, 'CausalForestDML': 1} 

{'tl': {'best_model_y': RandomForestRegressor(max_depth=10, n_estimators=25), 'best_model_t': '', 'Runtime': 0.14014148712158203}, 
'dml': {'best_model_y': MLPRegressor(max_iter=25), 'best_model_t': MLPClassifier(alpha=0.001, hidden_layer_sizes=(50,), max_iter=25), 'Runtime': 0.647728443145752}, 
'kernel_dml': {'best_model_y': RandomForestRegressor(max_depth=10, n_estimators=25), 'best_model_t': RandomForestClassifier(max_depth=5, min_samples_leaf=2, min_samples_split=5, n_estimators=25), 'Runtime': 0.319002628326416}, 
'CausalForestDML': {'best_model_y': RandomForestRegressor(max_depth=10, n_estimators=25), 'best_model_t': RandomForestClassifier(max_depth=5, min_samples_leaf=2, min_samples_split=5, n_estimators=25), 'Runtime': 0.4391043186187744}}


Before Cross-fitting

In [ ]:
key = 'ihdp'
before_crossfit_params_new.main(ci_estimator_list, model_y, model_t, key, k=4)

TWIN Dataset

During Cross-fitting

In [ ]:
key = 'twin'
during_crossfit_params.main(ci_estimator_list, model_y, model_t, key, k=4)

{'fold 0': {'model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'model_t': LogisticRegressionCV(max_iter=25), 'Mse': {'tl': 0.15256233788544776, 'dml': 0.10812668275965963, 'kernel_dml': 0.16572096128170893, 'CausalForestDML': 0.14120289983353243}, 'Runtime': {'tl': 0.3775370121002197, 'dml': 45.31202721595764, 'kernel_dml': 3.5571064949035645, 'CausalForestDML': 6.676959991455078}}, 

'fold 1': {'model_y': RandomForestRegressor(max_depth=10, n_estimators=25), 'model_t': RandomForestClassifier(min_samples_leaf=2, min_samples_split=5, n_estimators=25), 'Mse': {'tl': 0.16727746607929153, 'dml': 0.16571002180720207, 'kernel_dml': 0.17353813513727787, 'CausalForestDML': 0.16663956588132664}, 'Runtime': {'tl': 3.0297844409942627, 'dml': 244.7101273536682, 'kernel_dml': 5.239112138748169, 'CausalForestDML': 8.352089405059814}}, 

'fold 2': {'model_y': GradientBoostingRegressor(), 'model_t': GradientBoostingClassifier(learning_rate=0.01, n_estimators=500), 'Mse': {'tl': 0.16343999400854614, 'dml': 0.16666251940021218, 'kernel_dml': 0.17509369110091103, 'CausalForestDML': 0.1644531641068233}, 'Runtime': {'tl': 5.872559547424316, 'dml': 526.2035109996796, 'kernel_dml': 38.1200110912323, 'CausalForestDML': 41.825913190841675}}, 

'fold 3': {'model_y': MLPRegressor(alpha=0.01, max_iter=25), 'model_t': MLPClassifier(hidden_layer_sizes=(50,), learning_rate='adaptive', max_iter=25), 'Mse': {'tl': 5.339875644515088, 'dml': 850909599.5222945, 'kernel_dml': 0.18207610146862482, 'CausalForestDML': 1.386480025036902}, 'Runtime': {'tl': 7.452704668045044, 'dml': 62.683273792266846, 'kernel_dml': 13.718731164932251, 'CausalForestDML': 17.435981512069702}}} 

1722.3651785850525


{'tl': [0.15256233788544776, 0.16727746607929153, 0.16343999400854614, 5.339875644515088], 
'dml': [0.10812668275965963, 0.16571002180720207, 0.16666251940021218, 850909599.5222945], 
'kernel_dml': [0.16572096128170893, 0.17353813513727787, 0.17509369110091103, 0.18207610146862482], 
'CausalForestDML': [0.14120289983353243, 0.16663956588132664, 0.1644531641068233, 1.386480025036902]} 

{'tl': 0, 'dml': 0, 'kernel_dml': 0, 'CausalForestDML': 0} 

{'tl': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': '', 'Mse': 0.15256233788544776}, 
'dml': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Mse': 0.10812668275965963}, 'kernel_dml': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Mse': 0.16572096128170893}, 
'CausalForestDML': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Mse': 0.14120289983353243}}


{'tl': [0.3775370121002197, 3.0297844409942627, 5.872559547424316, 7.452704668045044], 
'dml': [45.31202721595764, 244.7101273536682, 526.2035109996796, 62.683273792266846], 
'kernel_dml': [3.5571064949035645, 5.239112138748169, 38.1200110912323, 13.718731164932251], 
'CausalForestDML': [6.676959991455078, 8.352089405059814, 41.825913190841675, 17.435981512069702]} 

{'tl': 0, 'dml': 0, 'kernel_dml': 0, 'CausalForestDML': 0} 

{'tl': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': '', 'Runtime': 0.3775370121002197}, 
'dml': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Runtime': 45.31202721595764}, 'kernel_dml': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Runtime': 3.5571064949035645}, 
'CausalForestDML': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Runtime': 6.676959991455078}}

Before Cross-fitting

In [ ]:
key = 'twin'
before_crossfit_params_new.main(ci_estimator_list, model_y, model_t, key, k=4)

ACIC Dataset

During Cross-fitting

In [3]:
key = 'acic'
during_crossfit_params.main(ci_estimator_list, model_y, model_t, key, k=4)

Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.10240448610767316, tolerance: 0.03355993749999999
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.06645613182831767, tolerance: 0.03355993749999999
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.03686042728429584, tolerance: 0.03363993749999999
lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org

{'fold 0': {'model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'model_t': LogisticRegressionCV(max_iter=25), 'Mse': {'tl': 0.2336780586051239, 'dml': 1.5811655289111919, 'kernel_dml': 0.25119735098996565, 'CausalForestDML': 0.24324824326252265}, 'Runtime': {'tl': 0.880598783493042, 'dml': 593.836683511734, 'kernel_dml': 5.199114561080933, 'CausalForestDML': 5.54042649269104}}, 'fold 1': {'model_y': RandomForestRegressor(max_depth=10, min_samples_split=10, n_estimators=25), 'model_t': RandomForestClassifier(max_depth=5, min_samples_split=5, n_estimators=25), 'Mse': {'tl': 0.2753620111579297, 'dml': 2.0738993420635023, 'kernel_dml': 0.26016960520601295, 'CausalForestDML': 0.23998113405233285}, 'Runtime': {'tl': 1.2489945888519287, 'dml': 426.3190679550171, 'kernel_dml': 1.7142417430877686, 'CausalForestDML': 2.37973690032959}}, 'fold 2': {'model_y': GradientBoostingRegressor(learning_rate=0.01), 'model_t': GradientBoostingClassifier(learning_rate=0.01), 'Mse': {'tl': 0.223393703918828,

Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.03067225236307536, tolerance: 0.02639466666666664
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.03262542865780915, tolerance: 0.02639466666666664
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.03285647873443054, tolerance: 0.02639466666666664
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.032133622479250334, tolerance: 0.02639466666666664
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.03287543811072169, tolerance: 0.02639466666666664
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.03538512744489708, tolerance: 0.02639466666666664
Objective did not converge. You might want to increase the number of iterations. Duality gap: 0.041090989847049286, toleran

{'fold 0': {'model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'model_t': LogisticRegressionCV(max_iter=25), 'Mse': {'tl': 0.2336780586051239, 'dml': 1.5811655289111919, 'kernel_dml': 0.25119735098996565, 'CausalForestDML': 0.24324824326252265}, 'Runtime': {'tl': 0.880598783493042, 'dml': 593.836683511734, 'kernel_dml': 5.199114561080933, 'CausalForestDML': 5.54042649269104}}, 

'fold 1': {'model_y': RandomForestRegressor(max_depth=10, min_samples_split=10, n_estimators=25), 'model_t': RandomForestClassifier(max_depth=5, min_samples_split=5, n_estimators=25), 'Mse': {'tl': 0.2753620111579297, 'dml': 2.0738993420635023, 'kernel_dml': 0.26016960520601295, 'CausalForestDML': 0.23998113405233285}, 'Runtime': {'tl': 1.2489945888519287, 'dml': 426.3190679550171, 'kernel_dml': 1.7142417430877686, 'CausalForestDML': 2.37973690032959}}, 

'fold 2': {'model_y': GradientBoostingRegressor(learning_rate=0.01), 'model_t': GradientBoostingClassifier(learning_rate=0.01), 'Mse': {'tl': 0.223393703918828, 'dml': 1.7726881599698143, 'kernel_dml': 0.25396002667772793, 'CausalForestDML': 0.2326347942638971}, 'Runtime': {'tl': 2.6335229873657227, 'dml': 984.3153734207153, 'kernel_dml': 5.502669334411621, 'CausalForestDML': 6.1991376876831055}}, 

'fold 3': {'model_y': MLPRegressor(alpha=0.01, max_iter=25), 'model_t': MLPClassifier(alpha=0.001, hidden_layer_sizes=(50,), learning_rate='adaptive', max_iter=25), 'Mse': {'tl': 738.4276637007682, 'dml': 579570357.0590864, 'kernel_dml': 2.2649587460827023, 'CausalForestDML': 14.698198490907634}, 'Runtime': {'tl': 0.7322561740875244, 'dml': 57.09786248207092, 'kernel_dml': 1.5269606113433838, 'CausalForestDML': 2.283345937728882}}} 

2361.8702080249786

{'tl': [0.2336780586051239, 0.2753620111579297, 0.223393703918828, 738.4276637007682], 
'dml': [1.5811655289111919, 2.0738993420635023, 1.7726881599698143, 579570357.0590864], 
'kernel_dml': [0.25119735098996565, 0.26016960520601295, 0.25396002667772793, 2.2649587460827023], 
'CausalForestDML': [0.24324824326252265, 0.23998113405233285, 0.2326347942638971, 14.698198490907634]} 
{'tl': 2, 'dml': 0, 'kernel_dml': 0, 'CausalForestDML': 2} 
{'tl': {'best_model_y': GradientBoostingRegressor(learning_rate=0.01), 'best_model_t': '', 'Mse': 0.223393703918828}, 
'dml': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Mse': 1.5811655289111919}, 
'kernel_dml': {'best_model_y': ElasticNetCV(l1_ratio=0.9, max_iter=100), 'best_model_t': LogisticRegressionCV(max_iter=25), 'Mse': 0.25119735098996565}, 
'CausalForestDML': {'best_model_y': GradientBoostingRegressor(learning_rate=0.01), 'best_model_t': GradientBoostingClassifier(learning_rate=0.01), 'Mse': 0.2326347942638971}}

{'tl': [0.880598783493042, 1.2489945888519287, 2.6335229873657227, 0.7322561740875244], 
'dml': [593.836683511734, 426.3190679550171, 984.3153734207153, 57.09786248207092], 
'kernel_dml': [5.199114561080933, 1.7142417430877686, 5.502669334411621, 1.5269606113433838], 
'CausalForestDML': [5.54042649269104, 2.37973690032959, 6.1991376876831055, 2.283345937728882]} 
{'tl': 3, 'dml': 3, 'kernel_dml': 3, 'CausalForestDML': 3} 
{'tl': {'best_model_y': MLPRegressor(alpha=0.01, max_iter=25), 'best_model_t': '', 'Runtime': 0.7322561740875244}, 
'dml': {'best_model_y': MLPRegressor(alpha=0.01, max_iter=25), 'best_model_t': MLPClassifier(alpha=0.001, hidden_layer_sizes=(50,), learning_rate='adaptive', max_iter=25), 'Runtime': 57.09786248207092}, 
'kernel_dml': {'best_model_y': MLPRegressor(alpha=0.01, max_iter=25), 'best_model_t': MLPClassifier(alpha=0.001, hidden_layer_sizes=(50,), learning_rate='adaptive',             max_iter=25), 'Runtime': 1.5269606113433838}, 
'CausalForestDML': {'best_model_y': MLPRegressor(alpha=0.01, max_iter=25), 'best_model_t': MLPClassifier(alpha=0.001, hidden_layer_sizes=(50,), learning_rate='adaptive',        max_iter=25), 'Runtime': 2.283345937728882}}





Before Cross-fitting

In [ ]:
key = 'acic'
before_crossfit_params_new.main(ci_estimator_list, model_y, model_t, key, k=4)